raiden/utils/signer.py

```diff
@@ -0,0 +1,87 @@
+from abc import ABC, abstractmethod
+from typing import Callable
+
+from eth_keys import keys
+from eth_keys.exceptions import BadSignature
+from eth_utils import keccak, to_checksum_address
+
+from raiden.exceptions import InvalidSignature
+from raiden.utils.typing import Address, AddressHex
+
+
+def eth_sign_sha3(data: bytes) -> bytes:
+    """
+    eth_sign/recover compatible hasher
+    Prefixes data with "\x19Ethereum Signed Message:\n<len(data)>"
+    """
+    prefix = b'\x19Ethereum Signed Message:\n'
+    if not data.startswith(prefix):
+        data = prefix + b'%d%s' % (len(data), data)
+    return keccak(data)
+
+
+def recover(
+        data: bytes,
+        signature: bytes,
+        hasher: Callable[[bytes], bytes] = eth_sign_sha3,
+) -> Address:
+    """ eth_recover address from data hash and signature """
+    _hash = hasher(data)
+
+    if signature[-1] >= 27:  # support (0,1,27,28) v values
+        signature = signature[:-1] + bytes([signature[-1] - 27])
+
+    try:
+        sig = keys.Signature(signature_bytes=signature)
+        public_key = keys.ecdsa_recover(message_hash=_hash, signature=sig)
+    except BadSignature as e:
+        raise InvalidSignature from e
+    return public_key.to_canonical_address()
+
+
+class Signer(ABC):
+    """ ABC for Signer interface """
+    # attribute or cached property which represents the address of the account of this Signer
+    address: Address
+
+    @abstractmethod
+    def sign(self, data: bytes, v: int = 27) -> bytes:
+        """ Sign data hash (as of EIP191) with this Signer's account """
+        pass
+
+    # TODO: signTransaction (replace privkey on JSONRPCClient)
+    # @abstractmethod
```

Comment:
Do you have any issues for this todo? Would be nice to have a reference as to who/when implements this TODO.

Reply:
Will create as soon as this PR is merged.

Reply:
I mean would be nice to reference the issue in the code ;)